kyo-core/shared/src/main/scala/kyo/Console.scala

```diff
@@ -80,12 +83,96 @@ object Console:
     def let[A, S](c: Console)(v: A < S)(using Frame): A < S =
         local.let(c)(v)
 
+    /** Executes a computation with access to the current Console instance.
+      *
+      * @param f
+      *   A function that takes a Console and returns a computation.
+      * @tparam A
+      *   The type of the computation result.
+      * @tparam S
+      *   The type of effects in the computation.
+      * @return
+      *   The result of the computation.
+      */
+    def use[A, S](f: Console => A < S)(using Frame): A < S =
+        local.use(f)
+
+    /** Gets the current Console instance from the local context.
+      *
+      * @return
+      *   The current Console instance.
+      */
+    def get(using Frame): Console < Any =
+        local.get
+
+    /** Executes a computation with a custom input stream containing the provided lines.
+      *
+      * @param lines
+      *   The lines to be used as input.
+      * @param v
+      *   The computation to execute.
+      * @tparam A
+      *   The type of the computation result.
+      * @tparam S
+      *   The type of effects in the computation.
+      * @return
+      *   The result of the computation with IO effects.
+      */
+    def withIn[A, S](lines: Iterable[String])(v: A < S)(using Frame): A < (IO & S) =
+        use { console =>
+            IO {
+                val it = lines.iterator
+                val proxy =
+                    new Proxy(console.unsafe):
+                        override def readln()(using AllowUnsafe) =
+                            if !it.hasNext then Result.fail(new EOFException("Consoles.readln failed."))
+                            else Result.success(it.next())
+                let(Console(proxy))(v)
+            }
+        }
+
+    /** Container for captured console output.
+      *
+      * @param stdOut
+      *   The captured standard output as a string.
+      * @param stdErr
+      *   The captured standard error output as a string.
+      */
+    case class Out(stdOut: String, stdErr: String)
+
+    /** Executes a computation while capturing its console output.
+      *
+      * @param v
+      *   The computation to execute.
+      * @tparam A
+      *   The type of the computation result.
+      * @tparam S
+      *   The type of effects in the computation.
+      * @return
+      *   A tuple containing the captured output (Out) and the computation result.
+      */
+    def withOut[A, S](v: A < S)(using Frame): (Out, A) < (IO & S) =
+        use { console =>
+            IO {
+                val stdOut = new StringBuffer
+                val stdErr = new StringBuffer
+                val proxy =
+                    new Proxy(console.unsafe):
+                        override def print(s: String)(using AllowUnsafe)      = Result.success(stdOut.append(s)).unit
```

Comment:
Will this actually output?

Reply:
No, the idea is to only collect the output